ApiType broke ALBs

caught from integ tests. This is broken because of the addition to http api (apigatewayv2) in deploylib

```
    index.ts(264,13): error TS2322: Type '{ Name: string; Routes: { execUnitName: string; path: string; verb: string; }[]; ApiType: string; }' is not assignable to type 'Gateway'.
      Object literal may only specify known properties, and '"ApiType"' does not exist in type 'Gateway'.
```

--- a/pkg/infra/pulumi_aws/deploylib.ts
+++ b/pkg/infra/pulumi_aws/deploylib.ts
@@ -29,6 +29,18 @@ export enum Resource {
     config = 'config',
 }
 
+export interface Gateway {
+    Name: string
+    Routes: Route[]
+    ApiType: 'REST' | 'HTTP'
+}
+
+export interface Route {
+    verb: string
+    path: string
+    execUnitName: string
+}
+
 export interface ResourceKey {
     Kind: string
     Name: string

--- a/pkg/infra/pulumi_aws/iac/api_gateway.ts
+++ b/pkg/infra/pulumi_aws/iac/api_gateway.ts
@@ -1,21 +1,9 @@
 import * as aws from '@pulumi/aws'
 import * as pulumi from '@pulumi/pulumi'
-import { CloudCCLib } from '../deploylib'
+import { CloudCCLib, Route, Gateway } from '../deploylib'
 import * as sha256 from 'simple-sha256'
 import { LoadBalancerPlugin } from './load_balancing'
 import { DeploymentArgs, StageArgs } from '@pulumi/aws/apigatewayv2'
-
-export interface Route {
-    verb: string
-    path: string
-    execUnitName: string
-}
-
-export interface Gateway {
-    Name: string
-    Routes: Route[]
-    ApiType: 'REST' | 'HTTP'
-}
 
 function sanitizeName(g: Gateway) {
     return g.Name.replace(/[^a-zA-Z0-9_-]+/g, '-')

--- a/pkg/infra/pulumi_aws/iac/load_balancing.ts
+++ b/pkg/infra/pulumi_aws/iac/load_balancing.ts
@@ -9,18 +9,7 @@ import {
 } from '@pulumi/aws/lb'
 import { ListenerRuleArgs } from '@pulumi/aws/alb'
 import { hash as h, sanitized } from './sanitization/sanitizer'
-import { CloudCCLib } from '../deploylib'
-
-export interface Route {
-    verb: string
-    path: string
-    execUnitName: string
-}
-
-export interface Gateway {
-    Name: string
-    Routes: Route[]
-}
+import { CloudCCLib, Gateway } from '../deploylib'
 
 export class LoadBalancerPlugin {
     // A map of all resources which are going to be fronted by a load balancer

